server/pages/index.js

```diff
@@ -120,13 +120,9 @@ function getCurrentCommitShortChecksum() {
 }
 
 function getDefaultContext( request ) {
-	let initialServerState = {};
-	// We don't cache routes with query params
-	if ( isEmpty( request.query ) ) {
```

Comment:
Starting to cache if there _are_ query params is a rather big change that should be highlighted in the PR desc, and deserves some discussion, since it means that cache slots will fill up way faster.